File: 노누리/1주차/10828.py
# input()으로 입력을 받았더니 입력 개수가 많아져서 시간초과가 발생한 문제
# sys.stdin.readline()으로 받아서 시간초과를 없애고
# 주피터로 돌리니까 너무 빨라서 입력을 받지도 않고 돌아가서 겨우 성공한 풀이

import sys
n=int(input())
stack=[]

def empty(stack):
    if len(stack)==0:
        return 1
    else:
        return 0

for _ in range(n):
    command=sys.stdin.readline().split()
    if len(command)==2:
        stack.append(int(command[1]))
    else:
        if command[0]=='pop':
            if empty(stack)==1:
                print('-1')
            else:
                print(stack.pop())
        elif command[0]=='size':
            print(len(stack))
        elif command[0]=='empty':
            print(empty(stack))
        elif command[0]=='top':
            if empty(stack)==1:
                print('-1')
            else:
                print(stack[-1])

File: 노누리/class2++/10828.java
package com.five;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Stack;
import java.util.StringTokenizer;

public class BJ10828 {
    public static void main(String[] args) throws IOException {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st=new StringTokenizer(br.readLine());
        StringBuilder sb=new StringBuilder();

        int N=Integer.parseInt(st.nextToken());
        Stack<Integer> stack=new Stack<>();
        for(int i=0;i<N;i++){
            st=new StringTokenizer(br.readLine());
            String[] com=st.nextToken().split(" ");

            if(st.hasMoreTokens()){
                stack.push(Integer.valueOf(st.nextToken()));
            }else{
                if(com[0].equals("pop")){
                    if(stack.isEmpty()){
                        sb.append("-1\n");
                    }else{
                        sb.append(stack.pop()+"\n");
                    }
                }else if(com[0].equals("size")){
                    sb.append(stack.size()+"\n");
                }else if(com[0].equals("empty")){
                    if(stack.isEmpty()){
                        sb.append("1\n");
                    }else{
                        sb.append("0\n");
                    }
                }else if(com[0].equals("top")){
                    if(stack.isEmpty()){
                        sb.append("-1\n");
                    }else{
                        sb.append(stack.peek()+"\n");
                    }
                }
            }
        }
        System.out.println(sb);
    }
}
